docs/types.rst

```diff
@@ -14,90 +14,279 @@ to form complex types.
+=============  ==========================================
+``x`` and ``y`` must be of the type ``decimal``. 
 
 .. _address:
-
 Address
--------
+=======
+**Keyword:** ``address``
 
-``address``: Holds an Ethereum address (20 byte value).
+The address type holds an Ethereum address.
 
+Values
+------
+An Address type can hold an Ethereum address which equates to 20 bytes/160 bits. Returns in hexadecimal notation with a leading ``0x``.
 
 .. _members-of-addresses:
+Members
+^^^^^^^
+
+============  ===================================================
+Member        Description
+============  ===================================================
+``balance``   Query balance of an address. Returns ``wei_value``.
+``codesize``  Query the code size of an address. Returns ``num``.
+============  ===================================================
+Syntax as follows: ``_address.<member>``, where ``_address`` is of the type ``address`` and ``<member>`` is one of the above keywords.
+
+Unit Types
+==========
+Viper allows the definition of types with a discrete unit such as e.g. meters, seconds, wei, ... . These types may only be based on either ``num`` or ``decimal``.
+Viper has multiple unit types built in, which are the following:
+
+=============  =====  =========  ==========================
+Time
+-----------------------------------------------------------
+Keyword        Unit   Base type  Description
+=============  =====  =========  ==========================
+``timestamp``  1 sec  ``num``    Represents a point in time
+``timedelta``  1 sec  ``num``    A number of seconds 
+=============  =====  =========  ==========================
 
-Members of Addresses
-^^^^^^^^^^^^^^^^^^^^
-
-* ``balance`` and ``send``
-
-It is possible to query the balance of an address using the property ``balance``
-and to send Ether (in units of wei) to an address using the ``send`` function:
-
-::
-
-    x: address
-
-    def foo(x: address):
-        if (x.balance < 10 and self.balance >= 10):
-            x.send(10)
-
+.. note::
+    Two ``timedelta`` can be added together, as can a ``timedelta`` and a ``timestamp``, but not two ``timestamps``.
+
+===================  ===========  =========  ====================================================================================
+Currency
+---------------------------------------------------------------------------------------------------------------------------------
+Keyword              Unit         Base type  Description
+===================  ===========  =========  ====================================================================================
+``wei_value``        1 wei        ``num``    An amount of `Ether <http://ethdocs.org/en/latest/ether.html#denominations>`_ in wei
+``currency_value``   1 currency   ``num``    An amount of currency
+``currency1_value``  1 currency1  ``num``    An amount of currency1
+``currency2_value``  1 currency2  ``num``    An amount of currency2
```

Comment:
@DavidKnott What the heck are `currency1_value` and `currency2_value` supposed to be?

Reply:
@fubuloubu `currency1_value` is a type to describe what the variable is and is checked on assignment so you can do something like:
```
total_cost: currency1_value
def foo(price: currenc1_value):
    self.total_cost = price
```
But not:
```
total_cost: currency2_value
def foo(price: currenc1_value):
    self.total_cost = price
```